orchagent/fgnhgorch.h

```diff
@@ -86,23 +94,30 @@ typedef struct
 class FgNhgOrch : public Orch, public Observer
 {
 public:
-    FgNhgPrefixes fgNhgPrefixes;
     FgNhgOrch(DBConnector *db, DBConnector *appDb, DBConnector *stateDb, vector<string> &tableNames, NeighOrch *neighOrch, IntfsOrch *intfsOrch, VRFOrch *vrfOrch);
 
     void update(SubjectType type, void *cntx);
-    bool addRoute(sai_object_id_t, const IpPrefix&, const NextHopGroupKey&);
-    bool removeRoute(sai_object_id_t, const IpPrefix&);
+    bool isRouteFineGrainedECMP(sai_object_id_t vrf_id, const IpPrefix &ipPrefix, const NextHopGroupKey &nextHops);
```

Comment:
>isRouteFineGrainedECMP [](start = 9, length = 22)

I am a little confused  about your naming conventions.

Are they the same thing?
- FineGrainedECMP
- FgNhg
- FineGrained
- m_fgNhgNexthops

Is it concise and understandable to call it `FineGroup` `isFineGroup`.

`isRouteFineGrainedECMP` -> `containsRoute` ?

Reply:
This is what each of them mean:
FineGrainedECMP: The term for the full feature -  Fine Grained ECMP
FgNhg: Fine Grained Next Hop Group. Similar in concept to ECMP next-hop group, but for Fine Grained ECMP behavior
FineGrained: Same as FineGrainedECMP, a concise name
m_fgNhgNexthops: The individual nexthops which comprise the FgNhg

Guess we can call it isFineGrained or isFgNhg

``containsRoute`` checks if a Fine Grained ECMP is associated with prefix is created in the ASIC
``isRouteFineGrainedECMP`` looks at current nhs of a route, and determines if it needs fine grained behavior. It is possible for isRouteFineGrainedECMP to be true while containsRoute is false, and all combinations of true/false. 

Can change to following, if you think its better ``isRouteFineGrainedECMP`` -> ``isNhgFineGrained`` , ``containsRoute`` -> ``containsSyncdFineGrained``.

Reply:
Changed code to isRouteFIneGrained and syncdContainsFgNhg. Let me know if this and above makes sense.